server/utils/match/occupancyCalendar.ts

```diff
@@ -0,0 +1,37 @@
+import { Cas1PremiseCapacity } from '@approved-premises/api'
+import { DateFormats } from '../dateUtils'
+import { dayAvailabilityCount } from './occupancy'
+
+type CalendarDay = {
+  name: string
+  bookableCount: number
+}
+type CalendarMonth = {
+  name: string
+  days: Array<CalendarDay>
+}
+type Calendar = Array<CalendarMonth>
+
+export const occupancyCalendar = (capacity: Cas1PremiseCapacity) => {
+  const calendar: Calendar = []
+
+  capacity.capacity.forEach(day => {
```

Comment:
This logic assumes that the calendar days are already sorted chronologically. I assume the API guarantees this.

Reply:
That's a good point, I'll check with API devs!

Reply:
Confirmed: dates are guaranteed chronological.